submissions/vl-shevchenko/object-oriented-javascript/app.js

```diff
@@ -0,0 +1,86 @@
+var Enemy = function (x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+
+    if (this.x > 510) {
+        this.x = -50;
+        this.speed = 100 + Math.floor(Math.random() * 222);
+    }
+
+    if (player.x < this.x + 80 &&
+        player.x + 80 > this.x &&
+        player.y < this.y + 60 &&
+        60 + player.y > this.y) {
+        player.x = 202;
+        player.y = 405;
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+let Player = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.moving = false;
+    this.sprite = 'images/char-boy.png';
+};
+
+Player.prototype.update = function () {
+
+};
+
+Player.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+
+};
+
+Player.prototype.handleInput = function (keyPress) {
+    if (keyPress == 'left' && this.x > 0) {
+        this.x -= 102;
+    }
+    if (keyPress == 'right' && this.x < 405) {
+        this.x += 102;
+    }
+    if (keyPress == 'up' && this.y > 0) {
+        this.y -= 83;
+    }
+    if (keyPress == 'down' && this.y < 405) {
+        this.y += 83;
+    }
+    if (this.y < 0) {
+        setTimeout(function () {
+            player.x = 202;
+            player.y = 405;
+        }, 600);
+    }
+}
+
+let allEnemies = [];
+let enemyLocation = [63, 147, 230];
```

Comment:
Mystery numbers. Here and across entire codebase.
See item 3 of the requirements